src/ucs/arch/x86_64/global_opts.c

```diff
@@ -17,28 +18,39 @@ ucs_config_field_t ucs_arch_global_opts_table[] = {
 #if ENABLE_BUILTIN_MEMCPY
   {"BUILTIN_MEMCPY_MIN", "auto",
    "Minimal threshold of buffer length for using built-in memcpy.",
-   ucs_offsetof(ucs_arch_global_opts_t, builtin_memcpy_min), UCS_CONFIG_TYPE_MEMUNITS},
+   ucs_offsetof(ucs_arch_global_opts_t, builtin_memcpy_min),
+   UCS_CONFIG_TYPE_MEMUNITS},
 
   {"BUILTIN_MEMCPY_MAX", "auto",
    "Maximal threshold of buffer length for using built-in memcpy.",
-   ucs_offsetof(ucs_arch_global_opts_t, builtin_memcpy_max), UCS_CONFIG_TYPE_MEMUNITS},
+   ucs_offsetof(ucs_arch_global_opts_t, builtin_memcpy_max),
+   UCS_CONFIG_TYPE_MEMUNITS},
 #endif
+  {"NT_BUFFER_TRANSFER_MIN", "auto",
+   "Minimal threshold of buffer length for using non-temporal buffer transfer.",
+   ucs_offsetof(ucs_arch_global_opts_t, nt_buffer_transfer_min),
+   UCS_CONFIG_TYPE_MEMUNITS},
   {NULL}
 };
 
 
 void ucs_arch_print_memcpy_limits(ucs_arch_global_opts_t *config)
 {
-#if ENABLE_BUILTIN_MEMCPY
     char min_thresh_str[32];
-    char max_thresh_str[32];
 
+#if ENABLE_BUILTIN_MEMCPY
+    char max_thresh_str[32];
     ucs_config_sprintf_memunits(min_thresh_str, sizeof(min_thresh_str),
                                 &config->builtin_memcpy_min, NULL);
     ucs_config_sprintf_memunits(max_thresh_str, sizeof(max_thresh_str),
                                 &config->builtin_memcpy_max, NULL);
-    printf("# Using built-in memcpy() for size %s..%s\n", min_thresh_str, max_thresh_str);
+    printf("# Using built-in memcpy() for size %s..%s\n",
+           min_thresh_str, max_thresh_str);
 #endif
-}
 
+    ucs_config_sprintf_memunits(min_thresh_str, sizeof(min_thresh_str),
+                                &config->nt_buffer_transfer_min, NULL);
+    printf("# Using nt-buffer-transfer for sizes from %s\n",
```

Comment:
do we need to also add  nt buffer dst threshold to the printed list? else it's not possible to know what size it is

Reply:
How to check these prints in a job started with mpirun? Any command-line param ?

Reply:
`mpirun -np 2 ucx_info -M` should do

Reply:
Done. Thanks for the command, it is printing as expected.